### PR TITLE
Propagate types through run_global_initializer

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: src/compiler/propagation/type_propagator.cc [hunk 1]
```diff
@@ -623,8 +648,14 @@ static void process(TypeScope* scope, uint8* bcp, std::vector<Worklist*>& workli
   OPCODE_END();
 
   OPCODE_BEGIN_WITH_WIDE(LOAD_GLOBAL_VAR_LAZY, index);
+    // The interpreter calls 'run_global_initializer_' to ensure
+    // that we deal with reentrant initialization correctly. This
+    // can throw.
+    propagator->ensure_run_global_initializer();
+    scope->throw_maybe();
+    // Analyze a call to the initialier method.
```
Comment (floitsch) [inline src/compiler/propagation/type_propagator.cc:656]:
```suggestion
    // Analyze a call to the initializer method.
    // The type of the global does not depend on the `run_global_initializer_` function,
    // and only on the actual initializer expression.
```

File: src/compiler/propagation/type_scope.cc [hunk 2]
```diff
@@ -21,6 +21,8 @@
 namespace toit {
 namespace compiler {
 
+static const int EXTRA = 1;
```
Comment (floitsch) [inline src/compiler/propagation/type_scope.cc:24]: Add a comment, what this constant is for.

Comment (kasperl) [reply src/compiler/propagation/type_scope.cc:27]: Done.

File: src/compiler/propagation/type_propagator.cc (hunk 1 above)
Comment (kasperl) [reply src/compiler/propagation/type_propagator.cc]: Done.

**Pull request merged**